src/socket.rs

```diff
@@ -382,3 +418,65 @@ impl From<mpsc::error::SendError<StateCommand>> for DisconnectError {
         DisconnectError::SocketShutdown
     }
 }
+
+#[derive(Clone)]
+struct ArcStatus {
+    status: Arc<AtomicU8>,
+    tx: broadcast::Sender<Result<Status, Arc<tungstenite::Error>>>,
```

Comment:
This would be more ergonomic as Arc<Result<Status, Error>>

Reply:
The `Arc` can't be lifted because the `tungstenite::Error` needs to be shared with the `ConnectError::WebSocketError` as discussed in https://github.com/liveview-native/phoenix-channels-client/pull/24#discussion_r1231271855.